js_modules/dagit/packages/core/src/workspace/asset-graph/AssetNode.tsx

```diff
@@ -13,26 +13,26 @@ import {
   Box,
 } from '@dagster-io/ui';
 import {isEqual} from 'lodash';
-import qs from 'qs';
-import React, {CSSProperties} from 'react';
-import {Link} from 'react-router-dom';
+import React from 'react';
 import styled from 'styled-components/macro';
 
-import {displayNameForAssetKey} from '../../app/Util';
+import {displayNameForAssetKey, withMiddleTruncation} from '../../app/Util';
 import {LATEST_MATERIALIZATION_METADATA_FRAGMENT} from '../../assets/LastMaterializationMetadata';
 import {NodeHighlightColors} from '../../graph/OpNode';
 import {OpTags} from '../../graph/OpTags';
 import {METADATA_ENTRY_FRAGMENT} from '../../metadata/MetadataEntry';
-import {titleForRun} from '../../runs/RunUtils';
 import {TimeElapsed} from '../../runs/TimeElapsed';
 import {TimestampDisplay} from '../../schedules/TimestampDisplay';
 import {buildRepoAddress} from '../buildRepoAddress';
-import {workspacePath, workspacePipelinePathGuessRepo} from '../workspacePath';
 
 import {LiveDataForNode, __ASSET_GROUP} from './Utils';
 import {AssetNodeFragment} from './types/AssetNodeFragment';
 import {useLaunchSingleAssetJob} from './useLaunchSingleAssetJob';
 
+const MAX_ANONTATIONS_WIDTH = 85;
+const DISPLAY_NAME_MAX_LENGTH = 50;
+const DISPLAY_NAME_PX_PER_CHAR = 8.0;
```

Comment:
It would be nice if we could make this feature more generic across monospace font usage and possibly add it to the core UI lib.